bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -248,23 +314,71 @@ protected NeoHubReadDcbResponse fromNeoHubReadDcbResponse() {
      * handlers
      */
     private synchronized void lazyPollingSchedulerExecute() {
-        NeoHubInfoResponse infoResponse = fromNeoHubReadInfoResponse();
+        // check which API is supported
+        if (!isApiOnline) {
+            selectApi();
+        }
 
-        if (infoResponse != null) {
-            // determine temperatureUnit
-            NeoHubReadDcbResponse dcbResponse = fromNeoHubReadDcbResponse();
-            Unit<?> temperatureUnit = (dcbResponse != null) ? dcbResponse.getTemperatureUnit() : SIUnits.CELSIUS;
+        NeoHubAbstractDeviceData deviceData = fromNeoHubGetDeviceData();
+        if (deviceData != null) {
+            // check the temperature unit
+            if (systemDataDirty) {
+                NeoHubReadDcbResponse systemData = fromNeoHubReadSystemData();
+                if (systemData != null) {
+                    temperatureUnit = systemData.getTemperatureUnit();
+                    systemDataDirty = false;
+                    logger.info("Setting temperature unit: => {}", temperatureUnit);
+                }
+            }
 
-            // dispatch infoResponse to each of the hub's owned devices ..
+            // dispatch deviceData to each of the hub's owned devices ..
             List<Thing> children = getThing().getThings();
             for (Thing child : children) {
                 ThingHandler device = child.getHandler();
                 if (device instanceof NeoBaseHandler) {
-                    ((NeoBaseHandler) device).toBaseSendPollResponse(infoResponse, temperatureUnit);
+                    ((NeoBaseHandler) device).toBaseSendPollResponse(deviceData);
                 }
             }
-        }
 
+            // evaluate and update the state of our RF mesh QoS channel
+            List<?> devices;
+            State state;
+
+            if ((devices = deviceData.getDevices()) == null) {
+                state = UnDefType.UNDEF;
+            } else {
+                int totalDeviceCount;
+
+                if ((totalDeviceCount = devices.size()) == 0) {
+                    state = UnDefType.UNDEF;
+                } else {
```

Comment:
Please don't perform assignments inside if conditionals if you could just initialize it when you declare the variable.
```suggestion
            State state;
            List<? extends AbstractRecord> devices = deviceData.getDevices();

            if (devices == null || devices.isEmpty()) {
                state = UnDefType.UNDEF;
            } else {
                int totalDeviceCount = devices.size();
```

Reply:
Ok.